### PR TITLE
added metamask check if user is connected for reality link

### PR DESCRIPTION
**Pull request by Mi-Lan**

### CODE REVIEW

File: app/src/components/market/common/additional_market_data/index.tsx [hunk 1]
```diff
@@ -64,10 +64,27 @@ interface Props extends DOMAttributes<HTMLDivElement> {
...
+        setConnected(accounts.length !== 0)
+      })
+    }
+    if (windowObj.ethereum) {
+      fetchUserAccounts()
+    }
+  }, [windowObj.ethereum])
   const realitioBaseUrl =
-    windowObj.ethereum && windowObj.ethereum.isMetaMask ? 'https://reality.eth' : 'https://reality.eth.link'
+    windowObj.ethereum && windowObj.ethereum.isMetaMask && connected
```
Comment (KadenZipfel) [inline app/src/components/market/common/additional_market_data/index.tsx:85]: Should be ok to remove `windowObj.ethereum && windowObj.ethereum.isMetaMask`, and just check that it's `connected`.

**Review by KadenZipfel — CHANGES_REQUESTED** [review]
This is good, but there are other places where we set the reality link which need to be updated. Bonus points if you can consolidate the reality link generating process to single place and display the generated link in each corresponding location.

File: app/src/components/market/common/additional_market_data/index.tsx (hunk 1 above)
Comment (Mi-Lan) [reply app/src/components/market/common/additional_market_data/index.tsx]: i implemented that in this new commit!

**Comment by Mi-Lan** [comment]
> This is good, but there are other places where we set the reality link which need to be updated. Bonus points if you can consolidate the reality link generating process to single place and display the generated link in each corresponding location.

Yeah that makes sense! I made new commit as you can see i made custom hook that encapsulates that logic and i just imported it everywhere i saw that reality link was used...

**Review by KadenZipfel — APPROVED** [review]
Good work 👏

**Pull request merged**